windows/ReactNativeCameraCPP/ReactCameraViewManager.cpp

```diff
@@ -0,0 +1,156 @@
+        nativeProps.Insert(L"flashMode", ViewManagerPropertyType::Number);
+        //        nativeProps.Insert(L"barcodeScannerEnabled", ViewManagerPropertyType::Boolean);
+        //        nativeProps.Insert(L"barCodeTypes", ViewManagerPropertyType::Array);
+        nativeProps.Insert(L"keepAwake", ViewManagerPropertyType::Boolean);
+
+        return nativeProps.GetView();
+    }
+
+    void ReactCameraViewManager::UpdateProperties(
+        FrameworkElement const& view,
+        IJSValueReader const& propertyMapReader) noexcept {
+        if (auto reactCameraView = view.try_as<ReactNativeCameraCPP::ReactCameraView>()) {
+            reactCameraView->UpdateProperties(propertyMapReader);
+        }
+    }
+
+    void ReactCameraViewManager::RemoveViewFromList(winrt::com_ptr<ReactNativeCameraCPP::ReactCameraView> view)
+    {
+        auto it = std::find(m_cameraViewInstances.begin(), m_cameraViewInstances.end(), view);
+        if (it != m_cameraViewInstances.end())
+            m_cameraViewInstances.erase(it);
+    }
+
+    IAsyncAction ReactCameraViewManager::TakePictureAsync(
+        std::map<std::wstring, winrt::Microsoft::ReactNative::JSValue> const& options,
+        int viewTag,
+        ReactPromise<JSValueObject>& result)
+    {
+        auto index = co_await FindCamera(viewTag);
+        if (index != -1)
+        {
+            auto cameraView = m_cameraViewInstances.at(index);
+            co_await cameraView->TakePictureAsync(options, result);
+        }
+        else
+        {
+            result.Reject("No camera instance found!");
+        }
+    }
+
+    IAsyncAction ReactCameraViewManager::RecordAsync(
+        std::map<std::wstring, JSValue> const& options,
+        int viewTag,
+        ReactPromise<JSValueObject>& result)
+    {
+        auto index = co_await FindCamera(viewTag);
+        if (index != -1)
+        {
+            auto cameraView = m_cameraViewInstances.at(index);
+            co_await cameraView->RecordAsync(options, result);
+        }
+        else
+        {
+            result.Reject("No camera instance found!");
+        }
+    }
+    
+    // Intialize MediaCapture and will bring up the constent dialog if this
+    // is the first time. RNC will display a progress indicator while waiting
+    // for user click. This method will resolve as no permission if user does
+    // not grant the permission or App does not have the appx capabilities for
+    // Microphone and WebCam specified in its Package.Appxmanifest.
+    IAsyncAction ReactCameraViewManager::CheckMediaCapturePermissionAsync(
+        winrt::Microsoft::ReactNative::ReactPromise<bool>& result)
+    {
+        auto mediaCapture = winrt::Windows::Media::Capture::MediaCapture();
+        bool hasPermission = true;
+        try {
+            co_await mediaCapture.InitializeAsync();
+        }
+        catch (winrt::hresult_error const&) {
+            hasPermission = false;
+        }

```

Comment:
should prob be a debug spew here

Reply:
Actually it is taken care by RN Camera, as it will show a "NotSupportedView" for RNCamera if this method returns no permission.